### ISSUE TITLE
MultipleNegativesSymmetricRankingLoss (potential optimization opportunity)

### ISSUE THREAD

**Issue by jangedoo** [UKPLab/sentence-transformers#2899]
In the forward method of [MultipleNegativesSymmetricRankingLoss](https://github.com/UKPLab/sentence-transformers/blob/add421f21508cd2baf4cd32af31624c63b355a1d/sentence_transformers/losses/MultipleNegativesSymmetricRankingLoss.py#L74)

```python
    def forward(self, sentence_features: Iterable[dict[str, Tensor]], labels: Tensor) -> Tensor:
        reps = [self.model(sentence_feature)["sentence_embedding"] for sentence_feature in sentence_features]
        anchor = reps[0]
        candidates = torch.cat(reps[1:])

        scores = self.similarity_fct(anchor, candidates) * self.scale
        labels = torch.tensor(
            range(len(scores)), dtype=torch.long, device=scores.device
        )  # Example a[i] should match with b[i]

        anchor_positive_scores = scores[:, 0 : len(reps[1])]
        ...
```

Any particular reason why we do `candidates = torch.cat(reps[1:])`? Couldn't we just do `candidates = reps[1]` since we just expect anchor and positive for this loss function and is much easier to read as well.

Also, in a dataset with many other columns besides anchor and positive, we unnecessarily compute similarity scores just to discard those scores later in this line `anchor_positive_scores = scores[:, 0 : len(reps[1])]`.

e.g. in a dataset with anchor, positive, negative columns with 10 rows and batch size of 10

anchor.shape =  10 x embed_dim
candidates.shape = 20 x embed_dim (first 10 will be from positive and second 10 from negative column)
scores.shape = 10 x 20
anchor_positive_scores.shape = 10 x 10

 What do you guys think?
